xcube/util/timeindex.py

```diff
@@ -111,6 +111,21 @@ def _ensure_timestamp_compatible(var: xr.DataArray, time_value: Any,
     if time_value is None:
         return None
 
+    if isinstance(time_value, np.ndarray):
+        if time_value.shape == ():
+            contents = time_value[()]
+            new_contents = _ensure_timestamp_compatible(
+                var, contents, time_name
+            )
+            return (
+                time_value if contents == new_contents
```

Comment:
In the current implementation they should be equivalent (since `_ensure_timestamp_compatible` only returns an equal time value when it actually returns the same object). But in general, if a future version of `_ensure_timestamp_compatible` returns a `new_contents` for which `(contents == new_contents) and (contents is not new_contents)` is true, then I would prefer to return the original `time_value` rather then unnecessarily creating a new array to represent the same value.